vllm/v1/core/sched/scheduler.py

```diff
@@ -317,9 +314,12 @@ def schedule(self) -> SchedulerOutput:
 
                 # P/D: skip request if still waiting for remote kvs.
                 if request.status == RequestStatus.WAITING_FOR_REMOTE_KVS:
-                    is_ready = self._update_waiting_for_remote_kv(
-                        request, skipped_waiting_requests)
-                    if not is_ready:
+                    is_ready = self._update_waiting_for_remote_kv(request)
+                    if is_ready:
+                        request.status = RequestStatus.WAITING
+                    else:
+                        self.waiting.popleft()
+                        skipped_waiting_requests.appendleft(request)
```

Comment:
Moved these bits out of the method to be more consistent with the handling of the other kinds of `WAITING_FOR_` states below (and this keeps all the request state changes clear in the main flow).